csr/csr.go

```diff
@@ -72,6 +75,15 @@ func (kr *KeyRequest) Generate() (crypto.PrivateKey, error) {
 			return nil, errors.New("RSA key size too large")
 		}
 		return rsa.GenerateKey(rand.Reader, kr.Size())
+	case "ed25519":
+		if kr.Size() != 256 && kr.Size() != 0 {
```

Comment:
If you omit the size parameter it will default to 0 on unmarshaling i think.
omitting it doesn't make sense for rsa or ecdsa, but ed25519 doesn't have parameters, so it looks cleaner to omit it in the json.

Reply:
`ed25519.SeedSize` return the size. Right now, seems to be checking for if the size is not 256 (which is correct) and not 0 (which is strange...)

Reply:
If it defaults to 0, then the condition should be an OR, I think..